bin/main.rb

```diff
@@ -1,74 +1,69 @@
 
-grilla = { 1 => 1, 2 => 2, 3 => 3, 4 => 4, 5 => 5, 6 => 6, 7 => 7, 8 => 8, 9 => 9 }
-SEPARATOR = '-------------'.freeze
-def create_grid(grilla)
-  puts ["| #{grilla[1]} | #{grilla[2]} | #{grilla[3]} |"]
-  puts SEPARATOR
-  puts ["| #{grilla[4]} | #{grilla[5]} | #{grilla[6]} |"]
-  puts SEPARATOR
-  puts ["| #{grilla[7]} | #{grilla[8]} | #{grilla[9]} |"]
-end
+require_relative '../lib/logic'
+
+class InitGame < WinCondition
+  @@grilla = [1, 2, 3, 4, 5, 6, 7, 8, 9]
 
-def do_move(move, _player2, grilla)
-  until ((1..9).include? move) && (grilla.value? move)
-    puts 'Invalid move'
-    puts 'Write another number from the board'
-    move = gets.chomp.to_i
+  def create_grid(arr)
+    puts "| #{arr[0]} | #{arr[1]} | #{arr[2]} |"
+    puts "| #{arr[3]} | #{arr[4]} | #{arr[5]} |"
+    puts "| #{arr[6]} | #{arr[7]} | #{arr[8]} |"
   end
-  grilla[move] = 'X'
-  create_grid(grilla)
-  puts "#{@player1} it's your turn. Choose one number from the board"
-end
 
-# loop
-def game_on(_player1, player2, grilla)
-  # call a method to verify if theres a winner when the conditions are accomplish or if it's a draw
-  # when the loop ends, ask the players if they want to play again or exit the game
-  win = false
-  draw = false
-  until win || draw
-    puts "#{@player1}, it's your turn, choose a number from the board"
-    move = gets.chomp
-    do_move(move, @player1, grilla)
-    puts "#{@player2}, it's your turn, choose a number from the board"
-    move = gets.chomp
-    do_move(move, player2, grilla)
-    create_grid(grilla)
-    draw = true
+  def dialogo
+    win_c = WinCondition.new
+
+    puts 'INSTRUCTIONS'
+    puts 'TIC TAC TOE'
+    puts 'Each player gets a turn'
+    puts "Don't try to cheat or you will get penalized"
+    puts 'You will loose a turn'
+    puts 'Whats your name?'
+
+    @player1 = gets.chomp
+    @player1 = gets.chomp while @player1 == ''
+    puts 'Whats the name of your partner?'
+    @player2 = gets.chomp
+    @player2 = gets.chomp while @player2 == ''
+    move = true
+    while @@grilla != []
+      if move == true
+        create_grid(@@grilla)
+        print "It is #{@player1} turn Pick a number : "
+        turn1 = gets.chomp
+        if turn1 =~ /^-?[1-9]+$/
+          turn1 = turn1.to_i
+          turn1 > 10 ? redo : turn1
+          win_c.change_element(turn1, @@grilla, 'X')
+        else
+          puts 'Please choose a number from 1-9'
+          redo
+        end
+        win_c.win(@@grilla, 'X', @player1)
+        move = false
+      else
+        create_grid(@@grilla)
+        print "It is #{@player2} turn Pick a number : "
+        turn2 = gets.chomp
+        if turn2 =~ /^-?[1-9]+$/
+          turn2 = turn2.to_i
+          turn2 > 10 ? redo : turn2
+          win_c.change_element(turn2, @@grilla, 'O')
+        else
+          puts 'Please choose a number from 1-9'
+          redo
```

Comment:
- [ ] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))